http/src/main/java/io/micronaut/http/bind/DefaultRequestBinderRegistry.java

```diff
@@ -93,6 +94,24 @@ public DefaultRequestBinderRegistry(ConversionService conversionService, Request
                 return () -> Optional.of(source);
             }
         });
+        byType.put(Argument.of(PushCapableHttpRequest.class).typeHashCode(), (RequestArgumentBinder<PushCapableHttpRequest>) (argument, source) -> {
+            if (source instanceof PushCapableHttpRequest) {
+                Optional<Argument<?>> typeVariable = argument.getFirstTypeVariable()
+                        .filter(arg -> arg.getType() != Object.class)
+                        .filter(arg -> arg.getType() != Void.class);
+                if (typeVariable.isPresent() && HttpMethod.permitsRequestBody(source.getMethod())) {
+                    if (source.getBody().isPresent()) {
+                        return () -> Optional.of(new PushCapableFullHttpRequest((PushCapableHttpRequest) source, typeVariable.get()));
+                    } else {
+                        return ArgumentBinder.BindingResult.UNSATISFIED;
```

Comment:
UNSATISFIED means that argument binding should be attempted multiple times. I think in this case if it fails the first time there is no point in retrying. If that is true, this should be EMPTY